activemodel/lib/active_model/type/helpers/time_value.rb

```diff
@@ -69,20 +69,36 @@ def new_time(year, mon, mday, hour, min, sec, microsec, offset = nil)
             \z
           /x
 
-          def fast_string_to_time(string)
-            return unless ISO_DATETIME =~ string
-
-            usec = $7.to_i
-            usec_len = $7&.length
-            if usec_len&.< 6
-              usec *= 10**(6 - usec_len)
+          if RUBY_VERSION >= "3.2"
```

Comment:
_[Probably non-blocking]_

I'm not sure how this is usually done on Rails, but should we use `Gem::Version` instead of plain strings in this comparison?

I know it's probably a silly use case, but `"10" > "3"` is false, for example.

```suggestion
          if Gem::Version.new(RUBY_VERSION) >= Gem::Version.new("3.2")
```

Reply:
@MatheusRich Yes, you're absolutely correct, in terms of theory. But with respect to the version number of this specific software called Ruby, 30 years of its history teaches us that the reality was that we needed 7 years to bump its major version from 2.0 (2013) to 3.0 (2020). With that in mind, we can deduce that we'll have to wait approximately 40 more years until we can start playing with Ruby 10. Perhaps we have to start considering about using `Gem::Version.new(RUBY_VERSION)` within the Ruby 9 era of 2060s, but until then, I prefer String comparisons which looks simpler for humans and runs faster on computers. :wink:

Reply:
That's what I expected to hear, and I'm okay with that. I was mostly curious to know what Rails usually do. Well, in 40 years we can talk about this again 😆

Reply:
```suggestion
          if Gem.ruby_version >= "3.2"
```